src/kontrol/foundry.py

```diff
@@ -784,6 +815,30 @@ def foundry_get_model(
     return '\n'.join(res_lines)
 
 
+def kevm_abstract_cells(cterm: CTerm, cells: Iterable[str]) -> CTerm:
+    kast = cterm.kast
+    var_count = count_vars(kast)
+
+    def _is_lone_var(_k: KInner) -> bool:
+        return type(_k) is KVariable and var_count[_k.name] == 1
+
+    for cell in cells:
+        cell_name = cell.upper() + '_CELL'
+        cell_contents = cterm.cell(cell_name)
+        new_cell_contents = cell_contents
+        if not _is_lone_var(cell_contents):
+            new_cell_contents = abstract_term_safely(kast, base_name=cell_name)
+            if cell_name == 'GAS_CELL' and type(cell_contents) is KApply and cell_contents.label.name == 'infGas':
+                if _is_lone_var(cell_contents.args[0]):
+                    new_cell_contents = cell_contents
```

Comment:
Does this say that if we already have just `#gas(VARIABLE)`, we don't do further changes? Why don't we do similar checks for other cells?

EDIT: We do do the checks, I just need to look at the code. I've understood the reasoning.